### PR TITLE
Refactor servlet 2

### PR DESCRIPTION
**Pull request by Mykhailenko**
Screenshot from 2019-09-02 11-13-00

### CODE REVIEW

File: rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/NodeSourceWindow.java [hunk 1]
```diff
@@ -685,14 +673,35 @@ private void populateHiddenItemsIfNecessary(PluginDescriptor pluginDescriptor) {
...
+            elasticInfra.setValue("Elastic");
+
+            FormItemIcon icon = new FormItemIcon();
+            icon.setSrc(RMImages.instance.good().getSafeUri().asString());
+            elasticInfra.setIcons(icon);
+            allFormItems.add(elasticInfra);
+        }
+    }
+
+    private String orderedFields(PluginDescriptor plugin) {
```
Comment (tao-xinxiu) [inline rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/NodeSourceWindow.java]: func can be renamed as `orderedNonFileFields` ?

File: rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/server/nodesource/NSCreationServlet.java [hunk 2]
```diff
@@ -271,6 +248,21 @@ private void createNs(HttpServletRequest request, HttpServletResponse response)
         }
     }
 
+    private void extractHiddenItems(Map<String, String> infraParams, Map<String, String> infraFileParams,
+            String formFieldValue) {
+        for (String row : formFieldValue.split(ROW_SEPARATOR)) {
+            String[] fields = row.split(FIELD_SEPARATOR);
```
Comment (tao-xinxiu) [inline rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/server/nodesource/NSCreationServlet.java]: Might we need to check `fields` length ?

File: rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/NodeSourceWindow.java (hunk 1 above)
Comment (Mykhailenko) [reply rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/NodeSourceWindow.java]: ok thanks

File: rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/server/nodesource/NSCreationServlet.java (hunk 2 above)
Comment (Mykhailenko) [reply rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/server/nodesource/NSCreationServlet.java]: ok thanks

**Pull request merged**